src/app/compiler.js

```diff
@@ -121,7 +117,7 @@ function Compiler (editor, handleGithubCall) {
       self.event.trigger('compilationFinished', [false, data, source])
     } else if (missingInputs !== undefined && missingInputs.length > 0) {
       // try compiling again with the new set of inputs
-      internalCompile(source.sources, missingInputs)
+      internalCompile(source.sources, source.target, missingInputs)
```

Comment:
This is the part I didn't finish. `source.target` must be added the compilation phase (it must be passed back).

Alternatively, we could wrap the whole compilation into a context (compiler instance) and have these there and not passed back and forth.